src/schemathesis/runner/__init__.py

```diff
@@ -89,9 +89,14 @@ def execute(  # pylint: disable=too-many-arguments
         hypothesis_options=hypothesis_options,
         loader=loader,
     )
-    all_events = list(generator)
-    finished = all_events[-1]
-    return finished.statistic
+    for r in generator:
```

Comment:
Do we need it as a generator? This one shouldn't be used by CLI, CLI should utilize `execute_as_generator`